Formula/telegram-cli.rb

```diff
@@ -0,0 +1,60 @@
+class TelegramCli < Formula
+  desc "Command-line interface for Telegram."
+  homepage "https://github.com/vysheng/tg"
+  # Use the tag instead of the tarball to get submodules
+  url "https://github.com/vysheng/tg.git",
+      :tag => "1.3.1",
+      :revision => "5935c97ed05b90015418b5208b7beeca15a6043c"
+  head "https://github.com/vysheng/tg.git"
+
+  depends_on "pkg-config" => :build
+  depends_on "readline"
+  depends_on "libevent"
+  depends_on "openssl"
+  depends_on "libconfig" => :recommended
+  depends_on "jansson" => :recommended
+  depends_on "lua" => :recommended
+  depends_on "python" => :recommended
+
+  patch :DATA
+
+  def install
+    args = %W[
+      --prefix=#{prefix}
+      CFLAGS=-I#{Formula["readline"].include}
+      CPPFLAGS=-I#{Formula["readline"].include}
+      LDFLAGS=-L#{Formula["readline"].lib}
+    ]
```

Comment:
okay, the problem is:
- the zlib check happens before the readline check
- the zlib check is dumb and adds `-I/usr/include` to CPPFLAGS
- SDKROOT is set, so /usr/include is not in the compiler's default search list, so -I/usr/include actually adds /usr/include to the top of the search list
- -I/usr/include appears on the command line before our -I flag for Homebrew's readline appears, so configure gets the wrong readline and barfs

We don't filter any flags for configure invocations so this is actually kind of annoying to fix.